packages/components/src/entities/SamplesTabbedGridPanel.spec.tsx

```diff
@@ -148,62 +167,111 @@ describe('SamplesTabbedGridPanel', () => {
             />
         );
         validate(wrapper);
+
         const buttonProps = wrapper.find(TabbedGridPanel).prop('buttonsComponentProps');
-        buttonProps['showBulkUpdate']();
-        wrapper.setProps({ actions: makeTestActions() }); // force re-render
+        act(() => {
+            buttonProps['showBulkUpdate']();
+            wrapper.setProps({actions: makeTestActions()}); // force re-render
+        });
+        await waitForLifecycle(wrapper);
         validate(wrapper, false, 'tab1', true);
 
         // call onBulkUpdateComplete with submitForEdit false, which does not puts the component in edit grid mode
-        wrapper.find(SamplesBulkUpdateForm).invoke('onBulkUpdateComplete')(fromJS({ a: 1 }), false);
-        wrapper.setProps({ actions: makeTestActions() }); // force re-render
+        act(() => {
+            wrapper.find(SamplesBulkUpdateForm).invoke('onBulkUpdateComplete')(fromJS({a: 1}), false);
+            wrapper.setProps({actions: makeTestActions()}); // force re-render
+        });
+        await waitForLifecycle(wrapper);
         validate(wrapper);
 
         // open back up the bulk edit modal
-        buttonProps['showBulkUpdate']();
-        wrapper.setProps({ actions: makeTestActions() }); // force re-render
+        act(() => {
+            buttonProps['showBulkUpdate']();
+            wrapper.setProps({actions: makeTestActions()}); // force re-render
+        });
+        await waitForLifecycle(wrapper);
 
         // call onBulkUpdateComplete with submitForEdit true, which puts the component in edit grid mode
-        wrapper.find(SamplesBulkUpdateForm).invoke('onBulkUpdateComplete')(fromJS({ a: 1 }), true);
-        wrapper.setProps({ actions: makeTestActions() }); // force re-render
+        act(() => {
+            wrapper.find(SamplesBulkUpdateForm).invoke('onBulkUpdateComplete')(fromJS({a: 1}), true);
+            wrapper.setProps({actions: makeTestActions()}); // force re-render
+        });
+        await waitForLifecycle(wrapper);
         validate(wrapper, true);
         wrapper.unmount();
     });
 
-    // Expected: Printing allowed
+
+    // Expected: Printing allowed on no tabs (Single model)
+    test('showLabelOption true, user.isGuest false, Single model', () => {
+        const wrapper = mountWithAppServerContext(<SamplesTabbedGridPanel {...SINGLE_TAB_PROPS} showLabelOption={true} />, {}, { user: TEST_USER_READER}, {}, {}, { printServiceUrl:"jest", labelTemplate:"jest"});
+        expect(wrapper.find(SamplesEditableGrid)).toHaveLength(0);
+        expect(wrapper.find(TabbedGridPanel)).toHaveLength(1);
+        expect(wrapper.find(TabbedGridPanel).prop('supportedExportTypes')).toBeDefined();
+        expect(wrapper.find(TabbedGridPanel).prop('onExport')).toBeDefined();
+        wrapper.unmount();
+    });
+
+    // Expected: Printing not allowed on single tab with Tabs always shown
+    test('showLabelOption true, user.isGuest false, Single model, alwaysShowTabs', () => {
```

Comment:
This relies on the assumption that the first tab is always the All Samples tab (which is not true here), so this will likely need adjusting.